### PR TITLE
Pagination through letters

### PR DESCRIPTION
**Pull request by vacas5**
Subdivided each letter into separate pages so we could make sure we
didn't miss any characters.

### CODE REVIEW

File: src/CharacterList.js [hunk 1]
```diff
@@ -8,46 +8,79 @@ class CharacterList extends Component {
...
     };
     this.fetchCharacters = this.fetchCharacters.bind(this);
+    this.resetPage = this.resetPage.bind(this);
+    this.incrementPage = this.incrementPage.bind(this);
+    this.decrementPage = this.decrementPage.bind(this);
   }
 
   componentDidMount() {
-    this.fetchCharacters(this.state.letter);
+    this.fetchCharacters(this.state.letter, 0);
```
Comment (nicrocs) [inline src/CharacterList.js]: Should we pass in this.state.offset here? or will it always start at 0?

File: src/CharacterList.js [hunk 2]
```diff
@@ -57,14 +90,19 @@ class CharacterList extends Component {
         <ul className="nav_items">
           {alphabet.map((letter, index) => <AlphaLinks letter={letter} key={index} selected={this.state.letter === letter} />)}
         </ul>
-        {!this.state.characters.length && <Loader />}
+        {!this.state.total && <Loader />}
```
Comment (nicrocs) [inline src/CharacterList.js]: Is there any potential for total to come back as zero or undefined from the fetch? if so, the loader won't disappear to display the error state

**Comment by nicrocs** [comment]
couple of small things

File: src/CharacterList.js (hunk 1 above)
Comment (vacas5) [inline src/CharacterList.js]: well, this.state.offset will always be zero at the start anyway

File: src/CharacterList.js (hunk 2 above)
Comment (vacas5) [inline src/CharacterList.js]: yeah I need to check on this if the request fails

Comment (vacas5) [inline src/CharacterList.js]: total always comes back from the request. If the request fails, we have other issues.

File: src/CharacterList.js (hunk 1 above)
Comment (vacas5) [inline src/CharacterList.js]: it needs to start at 0 if we are reloading the page, so I am going to leave it like this

**Pull request merged**